frontend/src/components/input.tsx

```diff
@@ -17,42 +19,59 @@ export interface InputProps {
 
 export const Input = ({ sendMessage, waiting }: InputProps) => {
   const [userInput, setUserInput] = useState<string>('');
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
 
-  const onChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+  const onChange = useCallback((event: ChangeEvent<HTMLTextAreaElement>) => {
     setUserInput(event.target.value);
   }, []);
 
+  useLayoutEffect(() => {
+    if (textareaRef.current) {
+      textareaRef.current.style.height = 'auto';
+      const textareaHeight = textareaRef.current.scrollHeight;
+
+      if (textareaHeight > 110) {
+        textareaRef.current.style.height = '110px';
+        textareaRef.current.style.overflowY = 'auto';
+      } else {
+        textareaRef.current.style.height = `${textareaHeight}px`;
+        textareaRef.current.style.overflowY = 'hidden';
+      }
+    }
+  }, [userInput]);
+
   const onSend = useCallback(
     (event: FormEvent<HTMLElement>) => {
       event.preventDefault();
-      sendMessage(userInput);
-      setUserInput('');
+      if (!waiting && userInput.trim().length > 0) {
+        sendMessage(userInput);
+        setUserInput('');
+      }
     },
-    [sendMessage, userInput],
+    [sendMessage, userInput, waiting],
   );
 
-  const sendDisabled = useMemo(() => userInput.length === 0, [userInput]);
-
   return (
     <>
       <form onSubmit={onSend} className={styles.inputContainer}>
-        <input
-          className={styles.input}
-          onChange={onChange}
-          onSubmit={onSend}
-          placeholder="Send a Message..."
-          type="text"
-          value={userInput}
-        />
-        <button
-          className={classNames(styles.sendButton, {
-            [styles.disabled]: sendDisabled,
-          })}
-          onClick={onSend}
-          disabled={sendDisabled}
-        >
-          <img src={RightArrow} />
-        </button>
+        <div className={styles.parentDiv}>
+          <textarea
+            className={styles.textarea}
+            ref={textareaRef}
+            placeholder="Send a Message..."
+            value={userInput}
+            onChange={onChange}
+            rows={1}
+          />
+          <div className={styles.uploadButton_container}>
+            <button className={styles.uploadButton} disabled={waiting}>
```

Comment:
clicking this button uploads, it should probably be disabled until this functionality is implemented